src/services/RewardService.ts

```diff
@@ -62,13 +69,22 @@ export default class RewardService {
 
         // Can not claim if the reward is disabled
         if (reward.state === RewardState.Disabled) {
-            return false;
+            throw new ForbiddenError('This reward already disabled by it owner');
+        }
+
+        // Can not claim if reward already extends the claim limit
+        // (included pending withdrawars)
+        if (reward.withdrawLimit > 0) {
+            const totalWithdrawed = await this.rewardProgress(reward);
+            if (totalWithdrawed >= reward.withdrawLimit) {
+                throw new ForbiddenError('This reward is reached it limit');
+            }
         }
 
         const withdrawal = await WithdrawalService.hasClaimedOnce(assetPool.address, account.id, reward.id);
         // Can only claim this reward once and a withdrawal already exists
         if (reward.isClaimOnce && withdrawal) {
-            return false;
+            throw new ForbiddenError('You have already claimed this reward');
```

Comment:
@GarfDev Any particular reason to change this to an HTTP error? The Wallet should be able to handle the reponse in order to display the correct view. Are you sure this does not break it? I will run a test for this.

Reply:
yes, I checked and see wallet behavior, it gets the error message from response and shows it to users, that is why I throw errors with messages to achieve this behaviour on wallet app
` Provide a specific error message in the 403 for every reason canClaim returns false. This should (might already do that) then show in Web Wallet.
`

Reply:
Ahh, thanks for your response and you are correct! Though we should not throw HTTP errors from services, since that would require the service call to always be invoked from a http controller and thats not always the case. E.g. jobs are not ran in this http context and might invoke service methods, resulting in dangling http errors.

I've changed the service to return error messages instead so the http controller can handle its response properly.